### PR TITLE
REST API: Ensure only requested resources are embedded

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Context
While working #8790, I noticed a core issue, which I reported [here](https://core.trac.wordpress.org/ticket/54015). We should fix the issue in our code. 

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [ ] I have added a matching `Type: XYZ` label to the PR

---

Fixes #

### CODE REVIEW

File: phpstan.neon.dist [hunk 1]
```diff
@@ -70,3 +70,6 @@ parameters:
 
     # False positive for wp_unslash()
     - '#^Cannot cast array\|string to string.$#'
+
+    # Core issue - https://core.trac.wordpress.org/ticket/54015
+    - '#^Parameter \#2 \$embed of method WP_REST_Server::envelope_response\(\) expects bool, array<string>\|bool given.$#'
```
Comment (swissspidy) [inline phpstan.neon.dist:75]: +1 to fixing the PHPDoc in core!

**Comment by swissspidy** [comment]
What issue does this fix for us? Any user-facing changes or other consequences?

**Comment by spacedmonkey** [comment]
> What issue does this fix for us? Any user-facing changes or other consequences?

It does. Here 
https://github.com/google/web-stories-wp/blob/20ade349e11c8d987363b74fbd056bc9b6530c87/includes/templates/admin/edit-story.php#L45
and here
https://github.com/google/web-stories-wp/blob/20ade349e11c8d987363b74fbd056bc9b6530c87/includes/Admin/Dashboard.php#L245

we embed requests in preloaded requests. ATM, passed, _embed=true, meaning all embeds are loaded. This fixes that and only embeds what is requests.

**Comment by spacedmonkey** [comment]
I found this issue while working on https://github.com/google/web-stories-wp/pull/8815

**Review by swissspidy — APPROVED** [review]

**Pull request merged**